backend/src/filter/EntityFilterConditionExecutor.ts

```diff
@@ -0,0 +1,65 @@
+import { Edge } from '../shared/entities/Edge';
+import { EdgeDescriptor } from '../shared/entities/EdgeDescriptor';
+import {
+  FilterCondition,
+  MatchAnyCondition,
+  MatchPropertyCondition,
+  OfTypeCondition,
+} from '../shared/queries';
+import FilterConditionVisitor from './FilterConditionVisitor';
+
+export class EdgeFilterConditionExecutor extends FilterConditionVisitor {
+  private edges: Edge[] = [];
+
+  public filterEdges(edges: Edge[], filter: FilterCondition): EdgeDescriptor[] {
+    this.edges = edges;
+    this.visit(filter);
```

Comment:
Currently, the check whether required properties are set or the filtering, is not implemented, right? So `visit()` does not do anything right now, correct?

Reply:
Also, I don't understand how `visit()` will filter the edges without getting them as a parameter.

Reply:
Of course `visit`is fully implemented. It is an instance method and has full access to `this.edges` that is set in line 15.